views/index.py

```diff
@@ -84,14 +86,23 @@ def registration():
         form = RegistrationForm()
         if form.validate_on_submit():
             try:
-                new_user = User(
-                    email=form.email.data,
-                    first_name=form.first_name.data,
-                    surname=form.surname.data,
-                    password_hash=generate_password_hash(form.password.data))
-                db.session.add(new_user)
-                db.session.commit()
-                send_confirmation_email(new_user.email)
+                data = User.query.filter_by(email=form.email.data).first()
+                if data is None:
```

Comment:
Are You sure "is None" is obligatory ?
Maybe:
if User.query.filter_by(email=form.email.data).first():
....
will be ok.

We can create user_exists(email) static method in User class.
if not User.user_exists(...):
...